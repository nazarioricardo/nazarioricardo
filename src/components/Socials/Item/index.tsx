import { StaticImageData } from "next/image";
import styles from "./styles.module.css";
import ImageFadeIn from "@nr/components/ImageFadeIn";

type ItemProps = {
  src: StaticImageData | string;
  link: string;
  alt: string;
};

function Item({ src, link, alt }: ItemProps) {
  return (
    <li className={styles.socialsItem}>
      <a href={link} target="_blank">
        <ImageFadeIn src={src} alt={alt} />
      </a>
    </li>
  );
}

export default Item;
