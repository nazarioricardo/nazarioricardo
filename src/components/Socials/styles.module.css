.container {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  height: 32px;
  width: "100%";
}
