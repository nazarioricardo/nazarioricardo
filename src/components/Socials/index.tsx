import Item from "./Item";
import Github from "./github.png";
import LinkedIn from "./linkedin.png";
import Instagram from "./instagram.png";
import styles from "./styles.module.css";

function Socials() {
  return (
    <ul className={styles.container}>
      <Item
        src={Instagram}
        link="https://www.instagram.com/ricrdo.n/"
        alt="Ricardo Nazario Instagram"
      />

      <Item
        src={LinkedIn}
        link="https://www.linkedin.com/in/ricardo-nazario-857b4951/"
        alt={"Ricardo Nazario LinkedIn"}
      />

      <Item
        src={Github}
        link="https://github.com/nazarioricardo"
        alt="Ricardo Nazario Github"
      />
    </ul>
  );
}

export default Socials;
