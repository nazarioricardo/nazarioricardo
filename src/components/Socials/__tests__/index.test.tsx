import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Socials from "../index";

describe("Socials", () => {
  test("renders Socials component", () => {
    render(<Socials />);
    const ulElement = screen.getByRole("list");
    expect(ulElement).toBeInTheDocument();
    expect(ulElement).toHaveClass("socials");
    const items = screen.getAllByRole("listitem");
    expect(items).toHaveLength(3);
    expect(screen.getByAltText("Ricardo Nazario Github")).toBeInTheDocument();
    expect(screen.getByAltText("Ricardo Nazario LinkedIn")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Socials />);
    expect(asFragment()).toMatchSnapshot();
  });
});
