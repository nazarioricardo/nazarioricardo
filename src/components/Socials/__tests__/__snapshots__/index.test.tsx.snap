// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Socials matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="socials"
  >
    <li
      class="socialsItem"
    >
      <a
        href="https://github.com/nazarioricardo"
        target="_blank"
      >
        <img
          alt="Ricardo Nazario Github"
          class="image false"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </a>
    </li>
    <li
      class="socialsItem"
    >
      <a
        href="https://www.linkedin.com/in/ricardo-nazario-857b4951/"
        target="_blank"
      >
        <img
          alt="Ricardo Nazario LinkedIn"
          class="image false"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </a>
    </li>
    <li
      class="socialsItem"
    >
      <a
        href="https://www.instagram.com/ricrdo.n/"
        target="_blank"
      >
        <img
          alt="Ricardo Nazario Instagram"
          class="image false"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
